### PR TITLE
MAGN-7016: OpenFileDialog

### PR DESCRIPTION
**Pull request by nguyen-binh-minh**
### Purpose

This PR adds a "Open in Manual Mode" checkbox in the OpenFileDialog so that the run setting can be override when opening a file.
openfiledialog

### Impact assessment

- [ ] Changes a public interface (OpenFile command)

#### Testing strategy

- Open dyn file with Automatic run setting and "Open in Manual Mode" unchecked, the run setting should be Automatic.
- Open dyn file with Automatic run setting and "Open in Manual Mode" checked, the run setting should be Manual.
- The state of the "Open in Manual Mode" should be the same the next time the OpenFileDialog is shown, i.e. if it's checked it should remain checked the next time it's shown.

### Reviewers

@sharadkjaiswal

### CODE REVIEW

File: src/DynamoCoreWpf/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -1682,4 +1682,7 @@ Do you want to install the latest Dynamo update?</value>
   <data name="ZeroTouchTypeShortString" xml:space="preserve">
     <value>DLL</value>
   </data>
+  <data name="FileDialogManualMode" xml:space="preserve">
+    <value>Open in Manual Mode</value>
```
Comment (sharadkjaiswal) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx]: I think the check box should say "Automatically execute after opening" or "Open in manual execution mode"

Comment (nguyen-binh-minh) [inline src/DynamoCoreWpf/Properties/Resources.en-US.resx]: updated

File: src/DynamoCore/Core/PreferenceSettings.cs [hunk 2]
```diff
@@ -148,6 +148,12 @@ public int MaxNumRecentFiles
         /// </summary>
         public bool PackageDownloadTouAccepted { get; set; }
 
+        /// <summary>
+        /// Indicates the default state of the "Open in Manual Mode"
+        /// checkbox in OpenFileDialog
+        /// </summary>
+        public bool ForceManualMode { get; set; }
```
Comment (sharadkjaiswal) [inline src/DynamoCore/Core/PreferenceSettings.cs]: Should be named as OpenInManualExecutionMode

File: src/DynamoCore/Models/RecordableCommands.cs [hunk 3]
```diff
@@ -459,6 +460,7 @@ internal static OpenFileCommand DeserializeCore(XmlElement element)
 
             [DataMember]
             internal string XmlFilePath { get; private set; }
+            internal bool ForceManualMode { get; private set; }
```
Comment (sharadkjaiswal) [inline src/DynamoCore/Models/RecordableCommands.cs]: ForceManualExecutionMode

File: src/DynamoCoreWpf/Interfaces/IShellCom.cs [hunk 4]
```diff
@@ -0,0 +1,502 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace Dynamo.Wpf.Interfaces
+{
+    internal static class IIDGuid
```
Comment (sharadkjaiswal) [inline src/DynamoCoreWpf/Interfaces/IShellCom.cs]: I assume these are all standard stuff obtained from some documentation. Can you please include the link here as comment?

File: src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs [hunk 5]
```diff
@@ -1093,14 +1094,26 @@ public FileDialog GetSaveDialog(WorkspaceModel workspace)
         /// Open a definition or workspace.
         /// </summary>
         /// <param name="parameters">The path the the file.</param>
+        /// <param name="forceManualMode"></param>
```
Comment (sharadkjaiswal) [inline src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs]: forceManualMode is not a parameter of Open()?

**Comment by sharadkjaiswal** [comment]
Apart from some minor comments, it looks good. @nguyen-binh-minh, thanks for working on it, great work.

**Comment by monikaprabhu** [comment]
@sharadkjaiswal - this needs to be merge into 0.8.1

**Pull request merged**